packages/amplify-graphql-index-transformer/src/graphql-index-transformer.ts

```diff
@@ -107,17 +108,24 @@ function validate(config: IndexDirectiveConfiguration, ctx: TransformerContextPr
     fieldMap.set(objectField.name.value, objectField);
 
     for (const peerDirective of objectField.directives!) {
+      const hasSortFields = peerDirective.arguments!.some((arg: any) => arg.name.value === 'sortKeyFields');
+
+      if (!defaultGSI && !hasSortFields && peerDirective.name.value === directiveName && objectField == config.primaryKeyField) {
```

Comment:
Does it work to move this validation down? The way this is currently written, it's possible that `config.primaryKeyField` won't be set yet. I'm also not sure that we need the `peerDirective.name.value === directiveName` part of this since we already know we are validating an `@index` directive.

Reply:
From what I understand, the `@index` shouldn't exist without a `sortKeyFields` when the `@index` is on the same field as a `@primaryKey`. So the only way `config.primaryKeyField` could be undefined would be if we had something like `foo: @index @primaryKey` would that be possible? Unfortunately, I cannot test it out myself since my AWS account is suspended right now.

Reply:
> `foo: @index @primaryKey` would that be possible?

Yes, I believe that would be possible.

> Unfortunately, I cannot test it out myself since my AWS account is suspended right now.

Since this only deals with input validation, you shouldn't need to deploy/test this in AWS.

Reply:
Thanks for the input, I made the changes and pushed it over again.

Reply:
Sorry, looks like moving it down fails quite a few tests. Do you think I should instead run a second `objectField.directives` loop for the validation after the current one so we know `config.primaryKeyField` will have to exist then?